sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusScaleMonitor.cs

```diff
@@ -57,23 +57,38 @@ async Task<ScaleMetrics> IScaleMonitor.GetMetricsAsync()
 
         public async Task<ServiceBusTriggerMetrics> GetMetricsAsync()
         {
-            ServiceBusReceivedMessage message = null;
+            ServiceBusReceivedMessage activeMessage = null;
             string entityName = _serviceBusEntityType == ServiceBusEntityType.Queue ? "queue" : "topic";
 
             try
             {
                 // Peek the first message in the queue without removing it from the queue
                 // PeekAsync remembers the sequence number of the last message, so the second call returns the second message instead of the first one
                 // Use PeekBySequenceNumberAsync with fromSequenceNumber = 0 to always get the first available message
-                message = await _receiver.Value.PeekMessageAsync(fromSequenceNumber: 0).ConfigureAwait(false);
+                do
+                {
+                    var receivedMessages = await _receiver.Value.PeekMessagesAsync(10, fromSequenceNumber: 0).ConfigureAwait(false);
```

Comment:
This will return the same 10 messages for each iteration of the loop since we always pass sequence number of 0. Even if we used the overload that allows the sequence number to advance, we could end up stuck in this loop for a really long time if there are millions of scheduled messages for example. I'd prefer to do a one shot peek, and if we only get scheduled/deferred messages, we can update the from sequence number so that next time we have a better chance.

Reply:
Are you sure that's true? At least it contradicts what is written in the comment here https://github.com/Azure/azure-sdk-for-net/pull/25127/files/035bb3870864f598edae3a58b5ff457379d559e9#diff-e2d09bf3bc95b6b03e0ddc0f67ea411f770344d9f1f598cd2dd9fad0bd0aa847L66. The behaviour of remembering the sequence number and passing 0 for fromSequenceNumber was something this code also previously relied upon. Or am I missing something?

Reply:
I have somewhere a sample I wrote weeks ago that verifies this. I will try to dig it out

Reply:
The comment is a bit confusing - I think it is saying that if you don't use the method that takes the sequence number it will keep incrementing, which was not the desired behavior.

https://github.com/Azure/azure-functions-servicebus-extension/commit/c72b7e4634b49fc296a2819365437f20ea4297dc

Reply:
Previously, we weren't looping and skipping scheduled/deferred, so always peeking from the head of the queue was what we wanted.

Reply:
I pushed a new version that does one peek and stores the sequence number for further retrieval